src/preact/base-element.js

```diff
@@ -755,49 +755,7 @@ function collectProps(Ctor, element, ref, defaultProps, mediaQueryProps) {
   }
 
   // Props.
-  for (const name in propDefs) {
-    const def = /** @type {!AmpElementPropDef} */ (propDefs[name]);
-    let value;
-    if (def.attr) {
-      value = element.getAttribute(def.attr);
-      if (def.media && value != null) {
-        value = mediaQueryProps.resolve(String(value));
-      }
-    } else if (def.parseAttrs) {
-      devAssert(def.attrs);
-      value = def.parseAttrs(element);
-    } else if (def.attrPrefix) {
-      const currObj = {};
-      let objContains = false;
-      const attrs = element.attributes;
-      for (let i = 0; i < attrs.length; i++) {
-        const attrib = attrs[i];
-        if (matchesAttrPrefix(attrib.name, def.attrPrefix)) {
-          currObj[dashToCamelCase(attrib.name.slice(def.attrPrefix.length))] =
-            attrib.value;
-          objContains = true;
-        }
-      }
-      if (objContains) {
-        value = currObj;
-      }
-    }
-    if (value == null) {
-      if (def.default != null) {
-        props[name] = def.default;
-      }
-    } else {
-      const v =
-        def.type == 'number'
-          ? parseFloat(value)
-          : def.type == 'boolean'
-          ? parseBooleanAttribute(/** @type {string} */ (value))
-          : def.type == 'date'
-          ? getDate(value)
-          : value;
-      props[name] = v;
-    }
-  }
+  parsePropDefs(props, propDefs, element, mediaQueryProps);
```

Comment:
This looks good, but we need a test for the nested parsing in the `test-base-element-mapping.js`

Reply:
Added unit tests, PTAL